GovUkDesignSystemComponents/Table.cshtml

```diff
@@ -1,49 +1,27 @@
-        $"govuk-table__caption "
-        + $"{(Model == null || String.IsNullOrEmpty(Model.CaptionClasses) ? string.Empty : Model.CaptionClasses)}";
-
-    List<TableCellViewModel> headerRow = new List<TableCellViewModel>();
-
-    if (Model.FirstCellIsHeader)
-    {
-        if (Model != null && Model.Rows != null && Model.Rows.Count > 0)
-        {
-            headerRow = Model.Rows[0].Row;
-        }
-    }
-    else
-    {
-        headerRow = Model.Head;
-    }
-}
 @if (Model != null)
 {
-    <table class="@classNames" @(Html.Raw(Model.Attributes != null ? string.Join(" ", Model.Attributes.Select(a => a.Key + "=\"" + a.Value + "\"")) : ""))>
+    <table class="govuk-table @Model?.Classes" @(Html.Raw(Model.Attributes != null ? string.Join(" ", Model.Attributes.Select(a => a.Key + "=\"" + a.Value + "\"")) : ""))>
 
-        @if (string.IsNullOrEmpty(Model.Caption))
+        @if (!string.IsNullOrEmpty(Model.Caption))
         {
-            <caption class="@Model.CaptionClasses">@Model.Caption</caption>
+            <caption class="govuk-table__caption @Model?.CaptionClasses">@Model.Caption</caption>
         }
 
-        @if (headerRow.Count > 0)
+        @if (Model.Head.Count > 0)
         {
             <thead class="govuk-table__head">
                 <tr class="govuk-table__row">
-                    @for (int cellIndex = 0; cellIndex < headerRow.Count; cellIndex++)
+                    @for (int cellIndex = 0; cellIndex < Model.Head.Count; cellIndex++)
                     {
-                        string colSpan = headerRow[cellIndex].Colspan >= 0 ? "colspan=" + headerRow[cellIndex].Colspan : string.Empty;
-                        string rowSpan = headerRow[cellIndex].Rowspan >= 0 ? "rowspan=" + headerRow[cellIndex].Rowspan : string.Empty;
-                        string tableHeadCellClasses = $"govuk-table__header " + $"{(!String.IsNullOrEmpty(headerRow[cellIndex].Format) && headerRow[cellIndex].Format.ToLower() == "mumeric" ? "govuk-table__header--numeric " : string.Empty)}" + $"{(String.IsNullOrEmpty(headerRow[cellIndex].Classes) ? string.Empty : headerRow[cellIndex].Classes)}";
-                        <th scope="col" class="@tableHeadCellClasses" @colSpan @rowSpan @(Html.Raw(headerRow[cellIndex].Attributes != null ? string.Join(" ", headerRow[cellIndex].Attributes.Select(a => a.Key + "=\"" + a.Value + "\"")) : ""))>
+                        string colSpan = Model.Head[cellIndex].Colspan != null ? "colspan=" + Model.Head[cellIndex].Colspan : string.Empty;
+                        string rowSpan = Model.Head[cellIndex].Rowspan != null ? "rowspan=" + Model.Head[cellIndex].Rowspan : string.Empty;
+                        string tableHeadCellClasses = $"govuk-table__header " + $"{(Model.Head[cellIndex].Format != null && Model.Head[cellIndex].Format == Format.Numeric ? "govuk-table__header--numeric " : string.Empty)}" + $"{(String.IsNullOrEmpty(Model.Head[cellIndex].Classes) ? string.Empty : Model.Head[cellIndex].Classes)}";
```

Comment:
Can we replace `$"{(String.IsNullOrEmpty(Model.Head[cellIndex].Classes) ? string.Empty : Model.Head[cellIndex].Classes)}"` with just `Model.Head[cellIndex].Classes` ?
Do we need `Model.Head[cellIndex].Format != null` does `Model.Head[cellIndex].Format == Format.Numeric` just work?

Reply:
Yes, we should replace and no I do not really need to check for null, I added it just because Format is null-able. Done, thanks